src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -8027,4 +8029,103 @@ void wm_vuldel_truncate_revision(char * revision) {
     return;
 }
 
+int wm_vuldet_recv_wdb_by_row(cJSON** j_array) {
+    char* response = NULL;
+    char* payload = NULL;
+    const char* error = NULL;
+    int size;
+    int parse_result;
+    int result = OS_INVALID;
+    bool sync_status = FALSE;
+    cJSON *j_object = NULL;
+    cJSON* j_status = NULL;
+
+    os_calloc(OS_MAXSTR, sizeof(char), response);
+    *j_array = cJSON_CreateArray();
+
+    do {
+        size = wm_vuldet_recv_wdb(response, OS_MAXSTR);
+        if (OS_INVALID == size) {
+            mdebug1("Error receiveing requested data from Wazuh-DB");
+            result = OS_INVALID;
+            break;
+        }
+        parse_result = wdbc_parse_result(response, &payload);
+        j_object = cJSON_ParseWithOpts(payload, &error, TRUE);
+        if (!j_object) {
+            mdebug1("Invalid JSON syntax in vulnerability detector database request.");
+            mdebug2("JSON error near: %s", error);
+            result = OS_INVALID;
+            break;
+        }
+
+        switch (parse_result) {
+            case WDBC_OK:
+                j_status = cJSON_GetObjectItem(j_object, "status");
+                if (cJSON_IsString(j_status)) {
+                    if (!strcmp(cJSON_GetStringValue(j_status),"SUCCESS")) {
+                        sync_status = TRUE;
+                        result = OS_SUCCESS;
+                    } else if ( !strcmp(cJSON_GetStringValue(j_status),"NOT_SYNCED")) {
+                        sync_status = FALSE;
+                        result = OS_SUCCESS;
+                    } else {
+                        sync_status = FALSE;
+                        result = OS_INVALID;
+                    }
+                } else {
+                    sync_status = FALSE;
+                    result = OS_INVALID;
+                }
+                break;
+            case WDBC_DUE:
```

Comment:
And one here explaining that a new itemwas received and there are pending ones.

Reply:
Fixed